secret/Tiltfile

```diff
@@ -0,0 +1,43 @@
+# -*- mode: Python -*-
+
+def secret_yaml_generic(name, namespace="", from_file=None):
+  """Returns YAML for a generic secret
+
+  Args:
+    name: The secret name.
+    namespace: The namespace.
+    from_file: Use the from-file secret generator. May be a string or a list of strings.
+       Example: ["ssh--privatekey=path/to/id_rsa", "ssh-publickey=path/to/id_rsa.pub"]
+
+  Returns:
+    The secret YAML as a blob
+  """
+  nsFlag = ""
+  if namespace:
+    nsFlag = " -n %s" % namespace
+
+  generator = ""
+  print(type(from_file))
+  if from_file:
+    if type(from_file) == "string":
+      generator = " --from-file=%s" % from_file
+    elif type(from_file) == "list":
+      generator = " ".join(["--from-file=%s" % s for s in from_file])
+    else:
+      fail("Bad from_file argument: %s" % from_file)
+
+  if not generator:
+    fail("No secret generator specified")
+
+  return local("kubectl create secret generic %s %s %s -o yaml --dry-run=client" % (name, nsFlag, generator))
```

Comment:
I think we should pass this as an array of strings to avoid shell problems.

I don't think proper escaping is super important for most cases (like, can you even put a space in a k8s secret name?), but it's more important in extensions:
1. if someone gets a value from some other function and passes it as a name here and it happens to contain a space, without escaping the error message might be very confusing (like instead of "invalid name" maybe there's a bunch of gobbledygook)
2. It's good to set a good example of proper input handling
3. It's good to get comfortable with it ourselves so that we can ensure that people who do care about it can do it (like maybe we actually do want a shell-escape builtin). If we find it's kind of annoying we should fix that!

Reply:
good point! i simply forgot we supported this. done!